MekHQ/unittests/mekhq/gui/dialog/nagDialogs/UnmaintainedUnitsNagDialogTest.java

```diff
@@ -0,0 +1,160 @@
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static mekhq.gui.dialog.nagDialogs.UnmaintainedUnitsNagDialog.checkHanger;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * This class is a test class for the UnmaintainedUnitsNagDialog class.
+ * It tests the different combinations of unit states and verifies the behavior of the checkHanger() method.
+ */
+class UnmaintainedUnitsNagDialogTest {
+    // Mock objects for the tests
+    Campaign campaign;
+    Hangar hangar;
+    Unit mockUnit1, mockUnit2;
+
+    /**
+     * Sets up the necessary dependencies and configurations before running the test methods.
+     * Runs once before all tests
+     */
+    @BeforeAll
+    static void setup() {
+        try {
+            Systems.setInstance(Systems.loadDefault());
+        } catch (Exception exception) {
+            MMLogger.create(UnmaintainedUnitsNagDialogTest.class).error("", exception);
+        }
+    }
+
+    /**
+     * Test setup for each test, runs before each test.
+     * Initializes the mock objects and sets up the necessary mock behaviors.
+     */
+    @BeforeEach
+    void init() {
+        // Initialize the mock objects
+        campaign = mock(Campaign.class);
+        hangar = mock(Hangar.class);
+        mockUnit1 = mock(Unit.class);
+        mockUnit2 = mock(Unit.class);
+
+        // When the Campaign mock calls 'getHangar()' return the 'hangar' mock
+        when(campaign.getHangar()).thenReturn(hangar);
+    }
+
+    /**
+     * Initializes the units by setting their maintenance status and salvage status.
+     *
+     * @param unit1Unmaintained A boolean indicating whether the first unit is unmaintained.
+     * @param unit1Salvage A boolean indicating whether the first unit is salvage.
+     * @param unit2Unmaintained A boolean indicating whether the second unit is unmaintained.
+     * @param unit2Salvage A boolean indicating whether the second unit is salvage.
+     */
+    private void initializeUnits(boolean unit1Unmaintained, boolean unit1Salvage, boolean unit2Unmaintained, boolean unit2Salvage) {
+        when(mockUnit1.isUnmaintained()).thenReturn(unit1Unmaintained);
+        when(mockUnit1.isSalvage()).thenReturn(unit1Salvage);
+
+        when(mockUnit2.isUnmaintained()).thenReturn(unit2Unmaintained);
+        when(mockUnit2.isSalvage()).thenReturn(unit2Salvage);
+
+        List<Unit> units = List.of(mockUnit1, mockUnit2);
+        when(hangar.getUnits()).thenReturn(units);
+    }
+
+    // In the following tests,
+    // Different combinations of unit states to set up desired behaviors in mock objects
+    // Then the checkHanger() method of UnmaintainedUnitsNagDialog class is called,
+    // and its response is checked against expected behavior
+
+    @Test
+    void unmaintainedUnitExistsUnit1() {
```

Comment:
This is all really good!  I have a single recommendation, not necessary to implement here but something to think about going forward:

Check out "parameterization", a method for reducing copy-and-pasting a bunch of similar tests (q.v. https://www.baeldung.com/parameterized-tests-junit-5).
When you have a set of tests that are identical except for some parameters that they consume, you can have Junit run the same test function with different combinations of parameters, thusly:

```
@ParameterizedTest
@ValueSource(ints = {1, 3, 5, -3, 15, Integer.MAX_VALUE})
void isOdd_ShouldReturnTrueForOddNumbers(int number) {
    assertTrue(Numbers.isOdd(number));
}
```

So this allows for running a test function multiple times with different inputs, cutting down on repetitive code.
There are drawbacks, though:

> One of the limitations of value sources is that they only support these types:
> short (with the shorts attribute)
> byte (bytes attribute)
> int (ints attribute)
> long (longs attribute)
> float (floats attribute)
> double (doubles attribute)
> char (chars attribute)
> java.lang.String (strings attribute)
> java.lang.Class (classes attribute)
> Also, we can only pass one argument to the test method each time.

Other things to note:
This only works if the assert is the same for every combo of parameters _or_ if you can pass in the expected value as an additional parameter... which is usually too much hassle.
So in _this_ case, you could parameterize all the `assertTrue` tests into one function, and all the `assertFalse` tests into another, if `initializeUnits()` could be set up to take a single value rather than 4.

Again, not requesting any changes here, just something you might want to look at for future work.

Reply:
This is very cool, thank you for the reading suggestion. I might be able to get it to work here if I can work out an algorithm that will allow us to test all the possible combinations, but I think in this instance what we would gain in brevity we'd lose in readability.

I know you said you weren't suggesting change, but I wanted to comment my thought process.